bokeh/core/property/instance.py

```diff
@@ -135,5 +136,14 @@ def _may_have_unstable_default(self):
 
 @register_type_link(Instance)
 def _sphinx_type_link(obj):
-    fullname = f"{obj.instance_type.__module__}.{obj.instance_type.__name__}"
-    return f"{property_link(obj)}({model_link(fullname)})"
+
+    # Sphinx links may be generated during docstring processing which means
+    # we can't necessarily evaluate pure function (e.g. lambda) Instance
+    # initializers, since they may contain circular references to the (not
+    # yet fully defined at this point) Model
+    if isinstance(obj._instance_type, types.FunctionType):
+        return f"{property_link(obj)}"
```

Comment:
The reason this was not an issue before is that signature/parameters processing was only applied to a subset of models that this circumstance did not happen to apply to.